includes/Core/Authentication/Credentials.php

```diff
@@ -54,6 +54,29 @@ public function __construct( Options $options ) {
 	 * @return array|bool Value set for the credentials, or false if not set.
 	 */
 	public function get() {
+		/**
+		 * Site Kit oAuth Secret is a JSON string of the Google Cloud Platform web application used for Site Kit
+		 * that will be associated with this account. This is meant to be a temporary way to specify the client secret
+		 * until the authentication proxy has been completed. This filter can be specified from a separate theme or plugin.
+		 *
+		 * To retrieve the JSON secret, use the following instructions:
+		 * - Go to the Google Cloud Platform and create a new project or use an existing one
+		 * - In the APIs & Services section, enable the APIs that are used within Site Kit
+		 * - Under 'credentials' either create new oAuth Client ID credentials or use an existing set of credentials
+		 * - Set the authorizes redirect URIs to be the URL to the oAuth callback for Site Kit, eg. https://<domainname>?oauth2callback=1 (this must be public)
+		 * - Click the 'Download JSON' button to download the JSON file that can be copied and pasted into the filter
+		 */
+		$credentials = json_decode( apply_filters( 'googlesitekit_oauth_secret', '' ), true );
```

Comment:
Preferably, let's only `json_decode()` this if it's not empty. Just saves us that step (and it will be empty in 99% of cases).

Reply:
I've updated this as requested. One added benefit here is that it also allows the array to be returned directly from the filter by only `json_decod`ing if it a non-empty string is returned. This is backwards compatible while allowing for the credentials to be filtered without _requiring_ it be a json string (which is a bit odd for a filter and should probably be deprecated/discouraged as the primary type). Thoughts?

Reply:
That sounds great. Preferably an array would have been nicer, but covering both variants is an elegant solution maintaining BC and also supporting a more proper format.